addons/mail/static/src/models/message_view.js

```diff
@@ -21,24 +21,43 @@ registerModel({
         /**
          * @param {MouseEvent} ev
          */
-        onClick(ev) {
+        async onClick(ev) {
             if (ev.target.closest('.o_channel_redirect')) {
-                this.messaging.openProfile({
+                // avoid following dummy href
+                ev.preventDefault();
+                let channel = this.messaging.models['Thread'].findFromIdentifyingData({
                     id: Number(ev.target.dataset.oeId),
                     model: 'mail.channel',
                 });
-                // avoid following dummy href
+                if(!channel) {
+                    channel = this.messaging.models['Thread'].insert({
+                        id: Number(ev.target.dataset.oeId),
+                        model: 'mail.channel',
+                    });
+                }
+                if(!channel.isPinned) {
+                    await channel.join();
```

Comment:
Note: this RPC might actually fail if the channel is no longer accessible to the current user, we might need to handle the error case.

Reply:
By default, it will display an ODOO `Acess Error`. So, I think it should be fine. And in case if the channel has been deleted by the admin. It shows a Missing Record Error.